File: Euclid/Messages/Incoming/CREATEFLAT.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;
using Euclid.Storage.Database.Data;
using System.Linq;

namespace Euclid.Messages.Incoming
{
    class CREATEFLAT : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            string name = request.GetArgument(2, "/");
            string model = request.GetArgument(3, "/");

            var roomModel = RoomManager.Instance.RoomModels.FirstOrDefault(x => x.Data.Model == model);

            if (roomModel == null)
                return;

            string modelType = roomModel.Data.Model.Replace("model_", "");

            if (modelType != "a" &&
                    modelType != "b" &&
                    modelType != "c" &&
                    modelType != "d" &&
                    modelType != "e" &&
                    modelType != "f")
            {
                return; // Fuck off, scripter.
            }

            RoomData roomData = new RoomData
            {
                OwnerId = player.Details.Id,
                Name = name,
                ModelId = roomModel.Data.Id,
                Description = string.Empty
            };

            RoomDao.NewRoom(roomData);

            player.Send(new FLATCREATED(roomData));
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/CREATEFLAT.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.room.RoomManager;
import euclid.game.room.RoomModel;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.FLATCREATED;
import euclid.network.streams.Request;
import euclid.storage.database.access.RoomDao;
import euclid.storage.database.data.RoomData;

public class CREATEFLAT implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        String name = request.getArgument(2, "/");
        String model = request.getArgument(3, "/");

        RoomModel roomModel = RoomManager.getInstance().getRoomModels().stream()
                .filter(x -> x.getData().getModel().equals(model))
                .findFirst()
                .orElse(null);

        if (roomModel == null) {
            return;
        }

        String modelType = roomModel.getData().getModel().replace("model_", "");

        if (!modelType.equals("a") &&
                !modelType.equals("b") &&
                !modelType.equals("c") &&
                !modelType.equals("d") &&
                !modelType.equals("e") &&
                !modelType.equals("f")) {
            return;
        }

        RoomData roomData = new RoomData();
        roomData.setOwnerId(player.getDetails().getId());
        roomData.setName(name);
        roomData.setModelId(roomModel.getData().getId());
        roomData.setDescription("");

        RoomDao.newRoom(roomData);

        player.send(new FLATCREATED(roomData));
    }
}
